### PR TITLE
[GH-22474] Support updating bot display name from app manifest

### PR DESCRIPTION
**Pull request by mujpao**
#### Summary

This pull request supports updating the display name and description of an app's associated bot account to match the app's display name when the app is reinstalled.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/22474

### CODE REVIEW

File: server/proxy/install.go [hunk 1]
```diff
@@ -263,6 +263,21 @@ func (p *Proxy) ensureBotNamed(r *incoming.Request, app *apps.App, icon io.Reade
 		} else {
 			bot.UserId = user.Id
 			bot.Username = user.Username
+
+			if user.GetDisplayName(model.ShowFullName) != app.DisplayName {
```
Comment (mickmister) [inline server/proxy/install.go:267]: I'm thinking we can have a comment here like this:

```suggestion
			// The app has changed its display name
			if user.GetDisplayName(model.ShowFullName) != app.DisplayName {
```

**Review by mickmister — APPROVED** [review]
Thanks for this improvement @mujpao!

LGTM, just one suggestion to include a comment here

**Review by hanzei — APPROVED** [review]

**Comment by mujpao** [comment]
I added a test in a new file. Is this the right way to do this?

**Review by mickmister — APPROVED** [review]

**Comment by hanzei** [comment]
Skipping QA testing as the changes are tested by e2e tests

**Pull request merged**